lib/check-or-update-routes.js

```diff
@@ -1,5 +1,7 @@
 module.exports = checkOrUpdateRoutes
 
+require('dotenv').config()
```

Comment:
instead of loading the configuration across multiple files, I’d prefer to have one file such as `lib/config.js` which loads it once and then the other files can do `const config = require('./config')`. Then we can also call the configuration in any way we want, e.g. `config.baseUrl` which we always use. The defaults will be defined in that file, too

Reply:
It would also be nice to allow to configure the options with CLI parameters, such as `--base-url`. That way we could run the `bin/octokit-rest-routes.js update` script twice.

Or when I think about it, we should handle all the environment variables / CLI arguments with `yargs` in `bin/octokit-rest-routes.js` – what do you think?

Reply:
>  I wonder if we could run this as part of our travis cron job?

I like this, but we'd need to establish a directory structure that would allow multiple versions of [`routes/routes`](https://github.com/octokit/routes/tree/master/routes).

Maybe something like the below?

```
.
├── enterprise-2.11.json
├── enterprise-2.12.json
├── enterprise-2.13.json
├── enterprise-2.14.json
├── index.json
└── routes
    ├── ALL-DOTCOM-ROUTES-LIVE-HERE
    └── enterprise
        ├── 2.11
        │   └── ALL-GHE2.11-ROUTES-LIVE-HERE
        ├── 2.12
        │   └── ALL-GHE2.12-ROUTES-LIVE-HERE
        ├── 2.13
        │   └── ALL-GHE2.13-ROUTES-LIVE-HERE
        └── 2.14
            └── ALL-GHE2.14-ROUTES-LIVE-HERE

```

> we should handle all the environment variables / CLI arguments with yargs in bin/octokit-rest-routes.js

So, we could potentially have a new command line argument, `--enterprise`, which when set to a GHE version, will _do the right thing_ for that environment, setting up the `BASE_URL` and `PATH_PREFIX` as needed.

So, in `.travis.yml` we might have something like:

```yaml
# ...
      script:
        - rm -rf index.json routes
        - bin/octokit-rest-routes.js update  # updates "dotcom" routes
        - rm -rf enterprise/2.11/index.json enterprise/2.11/routes
        - bin/octokit-rest-routes.js update --enterprise=2.11
        - rm -rf enterprise/2.12/index.json enterprise/2.12/routes
        - bin/octokit-rest-routes.js update --enterprise=2.12
        - rm -rf enterprise/2.13/index.json enterprise/2.13/routes
        - bin/octokit-rest-routes.js update --enterprise=2.13
        - rm -rf enterprise/2.14/index.json enterprise/2.14/routes
        - bin/octokit-rest-routes.js update --enterprise=2.14
# ...
```

This would mean though, that every new GitHub Enterprise release would require an update to this `.travis.yml`, is that OK do you think?

Reply:
I prepared a similar folder structure for octokit/fixtures: https://github.com/octokit/fixtures/tree/master/scenarios

Maybe let’s use `api.github.com` for the current routes, I think it’s very clear. And then folders for the currently supported GHE versions, such as ghe-2.11, etc. 

So the resulting structure would be 

```
routes/
├── api.github.com
├── ghe-2.11
├── ghe-2.12
├── ghe-2.13
└── ghe-2.14
```

The current `./index.json` would then live in `routes/api.github.com/index.json` (next to `activity.json`, `apps.json`, etc). 

> This would mean though, that every new GitHub Enterprise release would require an update to this .travis.yml, is that OK do you think?

Yeah let’s do that for now